PlaceRouteHierFlow/PnRDB/PnRdatabase.cpp

```diff
@@ -2170,66 +2170,86 @@ hierTree[i].Terminals[hierTree[i].Nets[j].connected[k].iter].netIter = j;
 }
 
 
+static string stem(const string& s) {
+
+  unsigned int start = 0;
+  unsigned int slash = s.find_last_of( '/');
+  if ( slash != string::npos) {
+    start = slash + 1;
+  }
+
+  unsigned int end = s.size();
+  unsigned int dot = s.find_last_of( '.');
+  if ( dot != string::npos) {
+    end = dot;
+  }
+
+  // xx/y.d
+  //   ^ ^
+  // 012345
+
+  return s.substr( start, end-start);
+}
+
 bool PnRdatabase::MergeLEFMapData(PnRDB::hierNode& node){
 
   auto logger = spdlog::default_logger()->clone("PnRDB.PnRdatabase.MergeLEFMapData");
 
   bool missing_lef_file = 0;
 
-  logger->info("merge LEF/map data on node {0}", node.name);
-  for(unsigned int i=0;i<node.Blocks.size();i++){
-    const string& master=node.Blocks[i].instance.back().master;
-    if(lefData.find(master)==lefData.end()) {
-	// LEF is missing; Ok if a cap or if not a leaf
-	if(master.find("Cap")!=std::string::npos ||
-	   master.find("cap")!=std::string::npos) continue;
-	if(node.Blocks[i].instance.back().isLeaf) {
-	    logger->error("The key does not exist in map: {0}",master);
-	    missing_lef_file = 1;
-	}
-	continue;
+  logger->info("merge LEF/map data on node {0}", node.name);  
+  for (unsigned int i = 0; i < node.Blocks.size(); i++) {
+    const string abstract_template_name = node.Blocks[i].instance.front().master;
+
+    if (gdsData2.find(abstract_template_name) == gdsData2.end()) {
+      if (abstract_template_name.find("Cap") != std::string::npos || abstract_template_name.find("cap") != std::string::npos || !node.Blocks[i].instance.back().isLeaf) continue;
```

Comment:
@854768750 I just fir checked if the abstract_template_name was in gdsData2. If not, I'll skip the whole process if cap is in the name.

Reply:
Currently this does not cause error on capacitor because capplacer is not functioning.

Reply:
This is fine because, seems in BUFFER_VREFP1, Dcap is just like other ordinary primitives.